crates/conversions/src/byte_array.rs

```diff
@@ -0,0 +1,42 @@
+use cairo_felt::Felt252;
+use cairo_lang_utils::byte_array::{BYTES_IN_WORD, BYTE_ARRAY_MAGIC};
+use itertools::chain;
+use num_traits::Num;
+
+pub struct ByteArray {
+    words: Vec<Felt252>,
+    pending_word_len: usize,
+    pending_word: Felt252,
+}
```

Comment:
I would avoid storing the internal Cairo implementation here, it is not guaranteed to be stable, even the compiler doesn't store this definition in their rust code. Instead I think we should pass it as an array of felts. Then we can convert it to sth printable on demand using functions exposed in the compiler just like Kamil did in #1645

Reply:
It's more for compatibility reasons than anything other, you can skip the serialization altogether in this way but it was easier just to follow the regular "cairo" way since it's efficient, verifiable (with appropriate serde + tests). User can also skip our deserialization and handle the serialized structure however he wants.

Reply:
Nvm, forgot we have to do `String` -> `Vec<Felt252>` conversion 🤦